include/hipSYCL/sycl/libkernel/cuda/group_functions.hpp

```diff
@@ -69,6 +74,11 @@ inline void group_barrier(sub_group g, memory_scope fence_scope) {
   }
   __syncwarp(); // not necessarily needed, but might improve performance
 }
+template<>
```

Comment:
Do we need the `template<>` here? Can we not make it a pure overload?

Reply:
We could go pure overload. Again I would apply it to all sub_group specialized functions, as above.